modules/contact/test/tests/bouncing-block-contact/frictionless-weighted-gap-mixed-basis.i

```diff
@@ -0,0 +1,173 @@
+starting_point = 2e-1
+offset = 1e-2
+
+[GlobalParams]
+  displacements = 'disp_x disp_y'
+  diffusivity = 1e0
+  scaling = 1e0
+[]
+
+[Mesh]
+  file = long-bottom-block-1elem-blocks.e
+  second_order = true
+  patch_update_strategy = always
+[]
+
+[Variables]
+  [./disp_x]
+    block = '1 2'
+    order = SECOND
+  [../]
+  [./disp_y]
+    block = '1 2'
+    order = SECOND
+  [../]
+  [./normal_lm]
+    block = 3
+  [../]
+[]
+
+[ICs]
+  [./disp_y]
+    block = 2
+    variable = disp_y
+    value = ${fparse starting_point + offset}
+    type = ConstantIC
+  [../]
+[]
+
+[Kernels]
+  [./disp_x]
+    type = MatDiffusion
+    variable = disp_x
+  [../]
+  [./disp_y]
+    type = MatDiffusion
+    variable = disp_y
+  [../]
+[]
+
+
+[Constraints]
+  [./weighted_gap_lm]
+    type = ComputeWeightedGapLMMechanicalContact
+    primary_boundary = 20
+    secondary_boundary = 10
+    primary_subdomain = 4
+    secondary_subdomain = 3
+    variable = normal_lm
+    secondary_variable = disp_x
```

Comment:
it is a bit confusing we do not call `secondary_disp_x`

Reply:
I have made changes so that the param name is now `disp_x`